packages/react-dom/src/events/EventListener.js

```diff
@@ -8,17 +8,33 @@
  */
 
 export function addEventBubbleListener(
-  element: Document | Element,
+  element: Document | Element | Node,
   eventType: string,
   listener: Function,
+  isPassive: boolean | null,
 ): void {
-  element.addEventListener(eventType, listener, false);
+  if (isPassive === null) {
+    element.addEventListener(eventType, listener, false);
+  } else {
+    element.addEventListener(eventType, listener, {
+      passive: isPassive,
+      capture: false,
+    });
```

Comment:
Good idea. Given the new API is still a while a way from being mainstream, it might be a hard limitation for this new API, as there appears to be no way to polyfill the behaviour.

Reply:
I'm not sure how we can test this at runtime to know if the various options are supported without adding quite a few more bytes. Is it worth it? We could just recommend a polyfill like we do in other places for those who need backwards support. https://github.com/WICG/EventListenerOptions/blob/gh-pages/EventListenerOptions.polyfill.js

Reply:
Looking at this polyfill I think it will be complicated to install it in apps that use React to render into a different JavaScript context (`<iframe>`, etc). I will have to do some testing to know for sure, though.

On MDN (https://developer.mozilla.org/de/docs/Web/API/EventTarget/addEventListener) it also seems that the `options` object is not implemented in Edge and IE11 which would affect quite a few users. I don't 100% trust this data though. If you want I can do some cross-browser tests to see which browsers support the options prop? 🙂 

---

> [..] know if the various options are supported 

I think that it is enough if we only test for passive event support. There are no other options that we would set. If passive events are not supported, we don't need the `options` prop.

Reply:
We could just fall back to not supplying options, which would allow `preventDefault` to work in all cases (not just nonPassive). What we could do, is set a flag to state this is the case and block `preventDefault()` on the SyntheticEvent object.

Reply:
Could we just capture all events?

Reply:
Here we go:
https://github.com/facebook/react/pull/12919

@philipp-spiess has a nice summary of the problems with my original attempt here:
https://github.com/facebook/react/pull/12919#issuecomment-395224674

Reply:
Won't capturing all events lead to the same issues pointed out in those threads? I'm happy to change it, if it makes sense to do so without any side-effects.

Reply:
Yes, sorry; I didn't give a very helpful comment either. 

At the time, I stopped work on this because it wasn't clear if there was a way forward. However I was curious if you had any ideas.

It might just be a dead end.